cmd/juju/application/deploy.go

```diff
@@ -1562,6 +1602,23 @@ func (c *DeployCommand) charmStoreCharm() (deployFn, error) {
 			return errors.Annotatef(err, "storing charm for URL %q", storeCharmOrBundleURL)
 		}
 
+		// If the original series was empty, so we couldn't validate the original
+		// charm series, but the charm url wasn't nil, we can check and validate
+		// what that one says.
+		//
+		// Note: it's interesting that the charm url and the series can diverge and
+		// tell different things when deploying a charm and in sake of understanding
+		// what we deploy, we should converge the two so that both report identical
+		// values.
+		if curl != nil && series == "" {
+			if err := c.validateCharmSeries(curl.Series); err != nil {
```

Comment:
I 've noticed that we are repeating this logic quite a few times. How about extracting it into a standalone method on `c`?

Reply:
Yeah, I'll clean this up, good shout